### PR TITLE
workflows/eval: Add the eval summary as a comment

### PR DESCRIPTION
**Pull request by azuwis**
See it in action https://github.com/azuwis/nixpkgs/pull/9

Make https://github.com/NixOS/nixpkgs/pull/360339 more discoverable.

Will post a comment like:

---

## [Eval summary](https://github.com/azuwis/nixpkgs/actions/runs/12116130895?pr=9)

- Added packages: 0
- Removed packages: 0
- Changed packages: 1
- Rebuild Linux: 1
- Rebuild Darwin: 0

---

Compared to https://github.com/NixOS/nixpkgs/pull/360962:

1) Only post single comment, and update it on subsequent push, reduce noise for long-running pull requests
    See https://github.com/NixOS/nixpkgs/pull/360962#discussion_r1865303075
2) Only summary and link, no details, will not affect GH search results
    See https://github.com/NixOS/nixpkgs/pull/360962#issuecomment-2510730459

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by FliegendeWurst — APPROVED** [review]

**Comment by FliegendeWurst** [comment]
I think the main benefit here is the link to the evaluation results. Numbers are already provided by the labels

File: .github/workflows/eval.yml [hunk 1]
```diff
@@ -267,3 +267,21 @@ jobs:
...
+          # Add the eval summary as a comment
+          body="$(jq -r '
+          "## [Eval summary](https://github.com/" + env.GH_REPO + "/actions/runs/" + env.GITHUB_RUN_ID + "?pr=" + env.NUMBER + ")\n\n" +
+          "- Added packages: " + (.attrdiff.added | length | tostring) + "\n" +
+          "- Removed packages: " + (.attrdiff.removed | length | tostring) + "\n" +
+          "- Changed packages: " + (.attrdiff.changed | length | tostring) + "\n" +
+          "- Rebuild Linux: " + (.rebuildCountByKernel.linux | tostring) + "\n" +
+          "- Rebuild Darwin: " + (.rebuildCountByKernel.darwin | tostring)
+          ' <comparison/changed-paths.json)"
+          gh pr comment "$NUMBER" --edit-last --body "$body" || gh pr comment "$NUMBER" --body "$body"
```
Comment (Mic92) [inline .github/workflows/eval.yml:283]: How is "last" selected? If `nixpkgs-review` posts an update via the api, would that count as the last comment?

Comment (Mic92) [reply .github/workflows/eval.yml:283]: Ah. No because the author is the nixos github app.

File: .github/workflows/eval.yml [hunk 2]
```diff
@@ -267,3 +267,21 @@ jobs:
           GH_TOKEN: ${{ github.token }}
           REPOSITORY: ${{ github.repository }}
           NUMBER: ${{ github.event.number }}
+
+      - name: Adding comments
```
Comment (Mic92) [inline .github/workflows/eval.yml:271]: I am a bit surprised that these permissions are enough here:

```
permissions:
  pull-requests: write
```

Comment (Mic92) [reply .github/workflows/eval.yml:271]: Ok. But looks like it's enough.

**Comment by Mic92** [comment]
https://github.com/NixOS/nixpkgs/pull/361344#issuecomment-2513701701

**Comment by infinisil** [comment]
Hmm not sure if I like having such a comment on every single PR

**Comment by azuwis** [comment]
How about only post the comment, if more than 10 packages are added/removed/changed? Like this https://github.com/NixOS/nixpkgs/pull/361365#issuecomment-2513813206

**Comment by Mic92** [comment]
Does requesting reviews actually creates a notification? I don't think it did this for me in the past. I agree with @infinisil I noticed that I found myself having to open updated pull requests more often just to see it was only yet another bot comment: https://github.com/NixOS/nixpkgs/pull/361388

**Pull request merged**